ocaml/xapi/xapi_pool.ml

```diff
@@ -190,13 +190,29 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
       raise (Api_errors.Server_error(Api_errors.pool_joining_host_cannot_contain_shared_SRs, []))
     end in
 
-  let assert_only_physical_pifs () =
-    let non_physical_pifs = Db.PIF.get_refs_where ~__context ~expr:(
+  let allow_management_interface_on_vlan () =
+    (* Allow pool-join if joining host and the pool have management interface on same vlan *)
+    match Db.VLAN.get_all ~__context with
+    | [] -> false
+    | [vlan] ->
+      (try
+        if Db.PIF.get_VLAN ~__context ~self:(
```

Comment:
You can replace  `if <expr> then true else false` with `expr`. You might want to let-bind some values to structure the `expr`.